server/src/test/java/org/elasticsearch/search/internal/ContextIndexSearcherTests.java

```diff
@@ -456,6 +458,57 @@ public TermsEnum iterator() {
         dir.close();
     }
 
+    public void testTimeExceedCallsReduce() throws IOException {
+        Directory dir = newDirectory();
+        RandomIndexWriter w = new RandomIndexWriter(random(), dir);
+        int docs = randomIntBetween(1, 1000);
+        for (int i = 0; i < docs; i++) {
+            Document doc = new Document();
+            StringField fooField = new StringField("foo", randomBoolean() ? "bar" : "foo", Field.Store.NO);
+            doc.add(fooField);
+            w.addDocument(doc);
+        }
+
+        ThreadPoolExecutor executor = (ThreadPoolExecutor) Executors.newFixedThreadPool(randomIntBetween(2, 5));
```

Comment:
Is it necessary to pass a thread pool? I was under the impression that that is not the case.

Reply:
This is order to test all possible code paths, concurrent (slices > 1) and not concurrent (slices == 0).

Reply:
sounds good, I randomized setting the executor, and I expanded the test to throw exception in either rewrite, createWeight or when pulling a scorer. Otherwise, we would always test the first one of these as the timeout runnable was registered before calling search.

Reply:
I also removed the indirection around registering the timeout runnable, in favour of simply throwing the exception in either rewrite, createWeight or scorer. I think that's simpler for this kind of test, as opposed to indirectly testing where the runnable is called as well.